File: Store/test/Sales.Domain.Tests/Generators/OrderItemGenerator.cs
using System;
using System.Collections.Generic;
using System.Text;
using FsCheck;
using Sales.Domain.Model.Orders;

namespace Sales.Domain.Tests.Generators
{
    public class OrderItemGenerator
    {
        public static Arbitrary<OrderItem> Generate()
        {
            return Arb.From(from id in Arb.Generate<long>()
                from quantity in Arb.Generate<PositiveInt>()
                from eachPrice in Arb.Generate<PositiveInt>()
                select new OrderItem(id, quantity.Get, eachPrice.Get));
        }
    }
}

File: src/test/java/store/sales/domain/generators/OrderItemGenerator.java
package store.sales.domain.generators;

import com.pholser.junit.quickcheck.generator.GenerationStatus;
import com.pholser.junit.quickcheck.generator.Generator;
import com.pholser.junit.quickcheck.random.SourceOfRandomness;
import store.sales.domain.model.orders.OrderItem;

public class OrderItemGenerator extends Generator<OrderItem> {
    private int MAX = Integer.MAX_VALUE;
    private int MIN = 0;

    public OrderItemGenerator() {
        super(OrderItem.class);
    }

    @Override
    public OrderItem generate(SourceOfRandomness random, GenerationStatus status) {
        PositiveIntGenerator positiveInt = gen().make(PositiveIntGenerator.class);
        positiveInt.configure(MIN, MAX);
        return new OrderItem(1,
                positiveInt.generate(random, status),
                positiveInt.generate(random, status));
    }
}
